commit-openapi-spec.yaml

```diff
@@ -0,0 +1,34 @@
+name: Update and Commit OpenAPI Spec
+on: [workflow_dispatch]
+jobs:
+  update-openapi-spec:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Install DJ
+        run: |
+          python -m pip install --upgrade pip
+          pip install .
+
+      - name: Generate OpenAPI Spec
+        run: |
+          python -c "
+          import json
+          from dj.api.query.engine import app
+
+          spec = app.openapi()
+          with open('openapi.json', 'w') as outfile:
+              outfile.write(json.dumps(spec, indent=4))
```

Comment:
Can we make this a script in dj/scripts/`, so we can run it manually if needed?

Reply:
Great idea! I added `scripts/generate-openapi.py` and call that in the workflow.